modules/cabal-project/ghcjs-overlay.nix

```diff
@@ -0,0 +1,31 @@
+{ lib, config, pkgs, ... }:
+let
+  cfg = config.ghcjs-overlay;
+in
+{
+  _file = "haskell.nix/modules/cabal-project/ghcjs-overlay.nix";
+
+  options = {
+    ghcjs-overlay.enable = lib.mkOption {
+      type = lib.types.bool;
+      default = pkgs.stdenv.hostPlatform.isGhcjs;
```

Comment:
Why are we doing this at all? Why can't we just get the user to add this if they want to build for ghcjs?

Reply:
Continuing on the thread [below](https://github.com/input-output-hk/haskell.nix/pull/2053#discussion_r1323030498).